### PR TITLE
Pull changes from `main` into `release/2.8`

### PR DESCRIPTION
**Pull request by AlanRynne**
## Description

Just your average update from `main` to `release/2.8` with a twist.

Since there were merge conflicts that needed to be resolved but **not included in `main`**, I created an auxiliary branch to fix these.

### CODE REVIEW

**Review by teocomi — APPROVED** [review]

**Comment by AlanRynne** [comment]
I think there's no need to wait for Code Owner approval (@clairekuang). changes in Rhino are not critical.

If Claire has an issue with any of these changes, please bring it up and we'll deal with it accordingly!

File: Objects/Converters/ConverterRevit/ConverterRevitShared/Partial Classes/ConvertFamilyInstance.cs
Comment (clairekuang) [inline Objects/Converters/ConverterRevit/ConverterRevitShared/Partial Classes/ConvertFamilyInstance.cs:155]: Logging should be updated to use the new applicationobject

Comment (AlanRynne) [reply Objects/Converters/ConverterRevit/ConverterRevitShared/Partial Classes/ConvertFamilyInstance.cs:155]: I created a separate issue for this so we don't forget! :D Good catch!

Comment (AlanRynne) [reply Objects/Converters/ConverterRevit/ConverterRevitShared/Partial Classes/ConvertFamilyInstance.cs:155]: https://github.com/specklesystems/speckle-sharp/issues/1473

**Pull request merged**
